packages/qwik/src/core/GLOSSARY.md

```diff
@@ -62,3 +62,90 @@ Here is the basic flow of data: JSX -> vDOM -> `vnode_diff` -> VNode -> DOM
 - `vnode_diff`: The algorithm that compares two vDOMs and produces a list of changes.
 - VNode: A layer on top of DOM which can be deserialized from SSR, and which understands Virtual Nodes and properties on those nodes.
 - DOM: The actual browser DOM.
+
+vNodes are Qwik's translation layer between your JSX and the actual HTML. They act like sticky notes, helping Qwik track and manage changes efficiently.
+
+```tsx
+<Component>
+  <button />
+</Component>
+```
+
+`Component` and `Signal` are examples of these sticky notes. Signal vNodes are particularly interesting as they create boundaries between normal text nodes, marking regions that can change.
+
+When Qwik updates your UI, it follows these steps:
+
+1. First, it looks at your sticky notes (vNodes) to see the current state
+2. Then, it compares this with your desired changes (JSX)
+3. Finally, it records the necessary changes in a journal, which will be applied to the HTML in a separate step
+
+## How the Diff Works
+
+The diff process uses two key pieces:
+
+- `jsxNode`: What you want your component to look like (input)
+- `vStartNode`: A snapshot of what your component currently looks like (output)
+
+> The diff only looks at the specific component or signal being updated, not the entire page.
+
+During this process, Qwik:
+
+1. Compares the current vNodes with your desired changes
+2. Records updates, additions, or removals in a change journal
+3. Handles cleanup of any removed elements
+4. Later applies these changes to ensure the final HTML matches what you wanted
+
+> On the server, Qwik only generates SSR nodes to create the initial HTML - vNodes only exist on the client where reconciliation is needed.
+
+### How Qwik handles props
+
+Qwik optimizes prop handling by distinguishing between two types:
+
+#### constProps (like JavaScript's `const`)
+
+The prop reference remains constant, though deep changes to objects/arrays are still possible through reactivity. For example, with a signal, the signal instance is constant, but its value can change.
+
+#### varProps (like JavaScript's `let`)
+
+The prop reference itself can change over time
+
+```tsx
+// constProp example with signal
+const count = useSignal(0);
+<Counter count={count} />; // signal instance is constant, but value changes
+
+// varProp example with store
+const store = useStore({ text: 'hello' });
+<Button text={store.text} />; // store property reference can change
```

Comment:
this is still const prop I think. if you do spread operation then for sure it will be var prop